packages/dialect-react-ui/components/common/ToggleSection.tsx

```diff
@@ -6,36 +6,46 @@ import { useTheme } from './providers/DialectThemeProvider';
 type PropsType = {
   title: string | React.ReactNode;
   className?: string;
-  enabled: boolean;
+  checked: boolean;
+  disabled?: boolean;
+  hideToggle?: boolean;
   onChange?: (nextValue: boolean) => void;
   children: React.ReactNode;
 };
 
 export default function ToggleSection({
   title,
-  enabled,
+  checked,
   children,
   onChange,
+  disabled,
+  hideToggle,
 }: PropsType) {
-  const [isEnabled, setEnabled] = useState(enabled);
+  const [isEnabled, setEnabled] = useState(checked);
   const { section } = useTheme();
 
   useEffect(() => {
-    setEnabled(enabled);
-  }, [enabled]);
+    setEnabled(checked);
+  }, [checked]);
 
   return (
     <div className={clsx(section)}>
-      <ValueRow className={clsx(isEnabled && 'dt-mb-2')} label={title}>
-        <Toggle
-          type="checkbox"
-          checked={isEnabled}
-          onClick={async () => {
-            const nextValue = !isEnabled;
-            await onChange?.(nextValue);
-            setEnabled(nextValue);
-          }}
-        />
+      <ValueRow
+        className={clsx(isEnabled && children && 'dt-mb-2')}
+        label={title}
+      >
+        {!hideToggle && (
+          <Toggle
+            type="checkbox"
+            checked={isEnabled}
+            disabled={disabled}
```

Comment:
Could you pls leave a todo to adjust the styles for a disabled toggle? Maybe even in the Toggle component itself.